src/test/suite/editors/codeActionProvider.test.ts

```diff
@@ -0,0 +1,89 @@
+import { beforeEach } from 'mocha';
+import chai from 'chai';
+
+import ActiveDBCodeLensProvider from '../../../editors/activeConnectionCodeLensProvider';
+import CodeActionProvider from '../../../editors/codeActionProvider';
+import ConnectionController from '../../../connectionController';
+import EditDocumentCodeLensProvider from '../../../editors/editDocumentCodeLensProvider';
+import { ExplorerController } from '../../../explorer';
+import { LanguageServerController } from '../../../language';
+import { PlaygroundController } from '../../../editors';
+import PlaygroundResultProvider from '../../../editors/playgroundResultProvider';
+import { StatusView } from '../../../views';
+import { StorageController } from '../../../storage';
+import TelemetryService from '../../../telemetry/telemetryService';
+import { TestExtensionContext, MockLanguageServerController } from '../stubs';
+
+const expect = chai.expect;
+
+suite('Code Action Provider Test Suite', () => {
+  const mockExtensionContext = new TestExtensionContext();
+
+  mockExtensionContext.extensionPath = '../../';
+
+  const mockStorageController = new StorageController(mockExtensionContext);
+  const testTelemetryService = new TelemetryService(
+    mockStorageController,
+    mockExtensionContext
+  );
+  const testStatusView = new StatusView(mockExtensionContext);
+  const testConnectionController = new ConnectionController(
+    testStatusView,
+    mockStorageController,
+    testTelemetryService
+  );
+  const mockLanguageServerController = new MockLanguageServerController(
+    mockExtensionContext,
+    mockStorageController
+  );
+  const testEditDocumentCodeLensProvider = new EditDocumentCodeLensProvider(
+    testConnectionController
+  );
+  const testPlaygroundResultProvider = new PlaygroundResultProvider(
+    testConnectionController,
+    testEditDocumentCodeLensProvider
+  );
+  const testActiveDBCodeLensProvider = new ActiveDBCodeLensProvider(
+    testConnectionController
+  );
+  const testExplorerController = new ExplorerController(
+    testConnectionController
+  );
+  let testPlaygroundController: PlaygroundController;
+
+  beforeEach(() => {
+    testPlaygroundController = new PlaygroundController(
+      mockExtensionContext,
+      testConnectionController,
+      mockLanguageServerController as LanguageServerController,
+      testTelemetryService,
+      testStatusView,
+      testPlaygroundResultProvider,
+      testActiveDBCodeLensProvider,
+      testExplorerController
+    );
+  });
+
+  test('expected provideCodeActions to return undefined when text is not selected', () => {
+    const testCodeActionProvider = new CodeActionProvider(testPlaygroundController);
+    const codeActions = testCodeActionProvider.provideCodeActions();
+
+    expect(codeActions).to.be.undefined;
+  });
+
+  test('expected provideCodeActions to return a run selected playground blocks action', () => {
```

Comment:
Would it be possible to add a test that actually runs the selected code? As far as I can tell, there wasn’t any before, but it might be a nice thing to have here